src/smt/seq_regex.cpp

```diff
@@ -90,6 +90,8 @@ namespace smt {
         expr* s = nullptr, *r = nullptr;
         expr* e = ctx.bool_var2expr(lit.var());
         VERIFY(str().is_in_re(e, s, r));
+        expr_ref _r(r, m);
```

Comment:
why do we need this?

Reply:
I added because I am worried about line 123, where we assign `r = re().mk_inter(r, s_approx);` but `r` is an expr* not an expr_ref so this does not save the result to make sure it is not deallocated. However I made a mistake -- `_r` needs to be set again to point to `r` right after line 123

Reply:
And also to guard against r and s being deallocated in `get_overapprox_regex`. Basically to ensure ownership over `r` and `s` for the duration of this function. However there may be a clearer way

Reply:
It may be best to just change r and s to `expr_ref`, then we don't have strange variables `_r` and `_s` to deal with. That may require some things to be changed to satisfy typechecking

Reply:
`e` owns r, s. So the reference is on e.

Reply:
`e` owns `r` but we later modify `r` without saving it. When we make `re().mk_inter` no one owns it.